<launch>

<arg name="markerId" default="0"/>
<arg name="markerSize" default="0.08"/> <!-- in meter -->
<arg name="eye" default="left"/>
<arg name="marker_frame" default="marker_frame"/>
<arg name="ref_frame" default="camera_link"/> <!-- leave empty and the pose will be published wrt param parent_name -->
<arg name="cam_frame" default="camera_color_optical_frame"/>
<arg name="camera" default="/camera/color"/>


<node name="image_proc" pkg="image_proc" type="image_proc"  ns="$(arg camera)">    
</node>

<node pkg="aruco_ros" type="single" name="aruco_single">
<remap from="/camera_info" to="$(arg camera)/camera_info" />
<remap from="/image" to="$(arg camera)/image_rect"/>
<param name="image_is_rectified" value="True"/>
<param name="marker_size" value="$(arg markerSize)"/>
<param name="marker_id" value="$(arg markerId)"/>
<param name="reference_frame" value="$(arg ref_frame)"/> <!-- frame in which the marker pose will be refered -->
<param name="camera_frame" value="$(arg cam_frame)"/>
<param name="marker_frame" value="$(arg marker_frame)" />

</node>

<node name="aruco_rgbd" pkg="visual_servoing" type="aruco_rgbd.py" output="screen"/>


</launch>

